File: typescript-stack/functions/list.ts
import { faas, documents } from "@nitric/sdk";
import { Example } from "../common";

// Start your function here
faas
  .http(async (ctx: faas.HttpContext): Promise<faas.HttpContext> => {
    try {
      const examples = await documents()
        .collection<Example>("examples")
        .query()
        .fetch();

      const exampleResults = [];

      for (const example of examples.documents) {
        exampleResults.push(example.content);
      }
      ctx.res.json(exampleResults);
    } catch (e) {
      ctx.res.status = 500;
      ctx.res.body = new TextEncoder().encode("An unexpected error occurred");
    }

    return ctx;
  })
  .start();

File: python-stack/functions/list.py
import json

from nitric.api.exception import NitricServiceException
from nitric.faas import start, HttpContext
from nitric.api import Documents


async def handler(ctx: HttpContext) -> HttpContext:
    try:
        examples_query = Documents().collection("examples").query()
        results = await examples_query.fetch()
        ctx.res.body = json.dumps([doc.content for doc in results.documents])
    except NitricServiceException:
        ctx.res.status = 500
        ctx.res.body = "An unexpected error occurred"

    return ctx


if __name__ == "__main__":
    start(handler)
